### ISSUE TITLE
no import suggestion

### ISSUE THREAD

**Issue by tianqing108** [astral-sh/ty#3191]
Description:
I am encountering an issue where ty fails to resolve local modules in my project. This seems to be a regression in recent versions, as the import suggestions were working previously.

Environment:

OS: Windows 11

IDE: VS Code (using the ty extension)

ty version: 0.0.27

Actual Behavior:
ty does not suggest the module name during typing and fails to resolve the path, leading to missing IntelliSense and false-positive diagnostic errors.

Expected Behavior:
<img width="2596" height="534" alt="Image" src="https://github.com/user-attachments/assets/47820860-2f4e-484d-86e1-73d5765b5a64" />

<img width="1104" height="736" alt="Image" src="https://github.com/user-attachments/assets/6e3ea8fb-742f-409f-a9a8-8dbc6309105c" />

<img width="1136" height="832" alt="Image" src="https://github.com/user-attachments/assets/c1a7d992-4024-4d0f-b284-646c17d89997" />

**Comment by tianqing108** [comment]
It seems that auto-import suggestions currently only work for classes within a module.

**Comment by MichaReiser** [comment]
Hmm weird. Can you change the [`logLevel`](https://docs.astral.sh/ty/reference/editor-settings/#loglevel) in your settings and share the output with us (Output panel -> ty language server)

**Comment by tianqing108** [comment]
@MichaReiser 

```
2026-04-02 11:02:50.086104500 DEBUG ty:worker:1 request{id=21 method="textDocument/hover"}: ty_ide::hover: Inferred type of covering node is Unknown
2026-04-02 11:02:50.086186100 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/hover" response.id=21 duration=276.70µs
2026-04-02 11:02:50.289087800 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/codeAction" response.id=22 duration=44.62ms
2026-04-02 11:02:50.996507300 DEBUG ty:worker:0 request{id=23 method="textDocument/hover"}: ty_ide::hover: Inferred type of covering node is Unknown
2026-04-02 11:02:50.996583800 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/hover" response.id=23 duration=178.80µs
2026-04-02 11:02:51.195407900 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/codeAction" response.id=24 duration=44.10ms
2026-04-02 11:02:59.150922900 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=25 duration=129.60µs
2026-04-02 11:02:59.846519900 DEBUG ty:worker:2 request{id=26 method="textDocument/hover"}: ty_ide::hover: Inferred type of covering node is Unknown
2026-04-02 11:02:59.846598700 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/hover" response.id=26 duration=205.20µs
2026-04-02 11:03:00.048379200 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/codeAction" response.id=27 duration=43.58ms
2026-04-02 11:03:00.480098700 DEBUG ty:worker:3 request{id=28 method="textDocument/hover"}: ty_ide::hover: Inferred type of covering node is Unknown
2026-04-02 11:03:00.480202000 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/hover" response.id=28 duration=241.60µs
2026-04-02 11:03:00.678435600 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/codeAction" response.id=29 duration=44.81ms
2026-04-02 11:03:01.512444400 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=30 duration=101.50µs
2026-04-02 11:03:03.772226700 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=31 duration=100.20µs
2026-04-02 11:03:04.330763100 DEBUG ty:worker:3 request{id=32 method="textDocument/hover"}: ty_ide::hover: Inferred type of covering node is Unknown
2026-04-02 11:03:04.330867000 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/hover" response.id=32 duration=223.00µs
2026-04-02 11:03:04.529985000 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/codeAction" response.id=33 duration=42.95ms
2026-04-02 11:03:05.859730400 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=34 duration=157.20µs
2026-04-02 11:03:07.347165700 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=35 duration=122.50µs
2026-04-02 11:03:08.062646200 DEBUG ty:worker:3 request{id=36 method="textDocument/hover"}: ty_ide::hover: Inferred type of covering node is Unknown
2026-04-02 11:03:08.062738700 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/hover" response.id=36 duration=207.90µs
2026-04-02 11:03:08.264376600 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/codeAction" response.id=37 duration=43.97ms
2026-04-02 11:03:08.424111700 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}: ty_server::server::api::notifications::did_change_watched_files: Applying changes to `c:\project\sports-server`
2026-04-02 11:03:08.424159300 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}:apply_changes: ty_project::db::changes: Handling file watcher change event: Changed { path: "c:\\project\\sports-server\\.git\\FETCH_HEAD", kind: Any }
2026-04-02 11:03:08.425342200 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/diagnostic" response.id=38 duration=88.50µs
2026-04-02 11:03:08.425577000 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/diagnostic" response.id=39 duration=102.20µs
2026-04-02 11:03:08.537881700 DEBUG ty:worker:3 request{id=40 method="textDocument/inlayHint"}: ty_server::server::api::requests::inlay_hints: Inlay hint request returned 0 hints in 57.2µs
2026-04-02 11:03:08.537942600 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/inlayHint" response.id=40 duration=177.70µs
2026-04-02 11:03:09.435301900 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=41 duration=146.30µs
2026-04-02 11:03:09.629311800 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}: ty_server::server::api::notifications::did_change_watched_files: Ignoring virtual path from change event: `git:/c%3A/project/sports-server/sites/fb/utils.py.git?%7B%22path%22%3A%22c%3A%5C%5Cproject%5C%5Csports-server%5C%5Csites%5C%5Cfb%5C%5Cutils.py%22%2C%22ref%22%3A%22%22%7D`
2026-04-02 11:03:09.629348600 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}: ty_server::server::api::notifications::did_change_watched_files: Ignoring virtual path from change event: `git:/c%3A/project/sports-server/test.py.git?%7B%22path%22%3A%22c%3A%5C%5Cproject%5C%5Csports-server%5C%5Ctest.py%22%2C%22ref%22%3A%22%22%7D`
2026-04-02 11:03:10.554778300 DEBUG     ty:main ty_server::server::main_loop: method="textDocument/documentHighlight" response.id=42 duration=111.80µs
```

**Comment by MichaReiser** [comment]
I don't see any completion requests. Can you make sure to restart VS Code, trigger some completions, and then share the logs?

**Comment by tianqing108** [comment]
@MichaReiser 

[log.txt](https://github.com/user-attachments/files/26432162/log.txt)

**Comment by MichaReiser** [comment]
Hmm, this is odd. I don't see any logged completion requests from VS Code. But looking at your example again, I'm not sure if this is indeed about completions. I think what you're trying to do is to import `llm` using VS Code's code actions. 

Your provided logs helped me to reproduce your code structure. If I didn't make any mistakes, it should be like so:



`test.py`:

```py
llm.Model

```

`logic/__init__.py`:

```py

```

`logic/llm.py`:

```py
class Model:
    ...
```

`logic/sibling.py`:

```py
llm.Model
```

[Playground](https://play.ty.dev/b89ff5af-c0f9-4312-bea0-fe71fa5e8079)

The reason ty fails to import llm is because the full import path from `test.py` to `llm` is `logic.llm` and not `llm`. ty does offer an import completion if you rewrite `test.py` to 



`test.py`:

```py
logic.llm.Model

```

An alternative is that you add `logic` to [`src.root`](https://docs.astral.sh/ty/modules/#first-party-modules), but you should only do this if it is the root where you store all your Python source code.

**Comment by tianqing108** [comment]
If I declare llm as exported in __init__.py, then import suggestions work correctly. Does this mean that ty’s design principle is that modules must be explicitly exported in order to get import completions, while classes do not need to be exported and still get completion suggestions?

```
from . import llm
 __all__ = ["llm"]

```

**Comment by MichaReiser** [comment]
Where did you place that `__init__.py`?
